packages/reactive_ble_mobile/ios/Classes/BleData extras/QualifiedCharacteristic.swift

```diff
@@ -13,8 +13,8 @@ extension QualifiedCharacteristic {
     init(_ characteristic: CBCharacteristic) {
         self.init(
             id: characteristic.uuid,
-            serviceID: characteristic.service.uuid,
-            peripheralID: characteristic.service.peripheral.identifier
+            serviceID: characteristic.service?.uuid ?? ServiceID(),
+            peripheralID: characteristic.service?.peripheral?.identifier ?? PeripheralID()
```

Comment:
While `ServiceID()` (which is an alias to `CBUUID()`) creates a zero-filled 16-bit ID, `PeripheralID()` (which is an alias to `UUID()`) creates a random UUID. The latter is for sure undesirable here.

Consider using (a long, but more suitable) `PeripheralID(uuid: (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0))`.

Reply:
fixed!